polars/polars-core/src/frame/hash_join/mod.rs

```diff
@@ -1287,7 +1287,6 @@ mod test {
             .unwrap();
         // create a new cache
         reset_string_cache();
-        let sc = IUseStringCache::new();
```

Comment:
I think we need to underscore the `sc` as `_sc`. This lock is needed to ensure we hold the string cache.

Reply:
Does what 7d09e46b627b9f28e9010f75246c846813bfdea9 proposes works for you?

(I mis-clicked and thus have merged it as is without a proper commit message, but I guess PR are squashed prior merges?)

Reply:
Yeap, that works. :+1: